dojo/utils.py

```diff
@@ -2108,3 +2121,102 @@ def is_scan_file_too_large(scan_file):
+    Exception is for JIRA issues, which would get a comment anyways.
+    """
+    def _notify(finding, title):
+        create_notification(
+            event='sla_breach',
+            title=title,
+            finding=finding,
+            sla_age=sla_age
+        )
+
+        if do_jira_sla_comment:
+            logger.info("Creating JIRA comment to notify of SLA breach information.")
+            add_simple_jira_comment(jira_config, jira_issue, title)
+
+    # exit early on flags
+    if not settings.SLA_NOTIFY_ACTIVE and not settings.SLA_NOTIFY_ACTIVE_VERIFIED_ONLY:
+        logger.info("Will not notify on SLA breach per user configured flags")
+        return
+
+    jira_issue = None
+    jira_config = None
+    try:
+        system_settings = System_Settings.objects.get()
+        if system_settings.enable_finding_sla:
+            logger.info("About to process findings for SLA notifications.")
+            logger.debug("Active {}, Verified {}, Has JIRA {}, pre-breach {}".format(
+                settings.SLA_NOTIFY_ACTIVE,
+                settings.SLA_NOTIFY_ACTIVE_VERIFIED_ONLY,
+                settings.SLA_NOTIFY_WITH_JIRA_ONLY,
+                settings.SLA_NOTIFY_PRE_BREACH,
+            ))
+
+            query = None
+            if settings.SLA_NOTIFY_ACTIVE:
+                query = Q(active=True, is_Mitigated=False, duplicate=False)
+            if settings.SLA_NOTIFY_ACTIVE_VERIFIED_ONLY:
+                query = Q(active=True, verified=True, is_Mitigated=False, duplicate=False)
+            logger.debug("My query: {}".format(query))
+
+            no_jira_findings = {}
+            if settings.SLA_NOTIFY_WITH_JIRA_ONLY:
+                logger.debug("Ignoring findings that are not linked to a JIRA issue")
+                no_jira_findings = Finding.objects.exclude(jira_issue__isnull=False)
+            findings = Finding.objects \
+                .filter(
+                    query
+                ).exclude(id__in=no_jira_findings)
+
+            for finding in findings:
+                # A finding with 'Info' severity will not be considered for SLA notifications (not in model)
+                if finding.severity == 'Info':
+                    continue
+
+                sla_age = finding.sla_days_remaining()
+                # if SLA is set to 0 in settings, it's a null. And setting at 0 means no SLA apparently.
+                if sla_age is None:
+                    sla_age = 0
+
+                do_jira_sla_comment = False
+                if finding.has_jira_issue():
+                    jira_config = finding.jira_conf_new()
+                    if jira_config is not None:
+                        logger.debug("JIRA config for finding is {}".format(jira_config))
+                        # global config or product config set, product level takes precedence
+                        product_jira_sla_comment_enabled = finding.test.engagement.product.jira_pkey_set.first().product_jira_sla_notification
```

Comment:
this could crash if there is no jira_jpkey for a product (corner case).
in my JIRA PR I have introduced a `jira_pkey` property for product that will work with prefetching.